internal/providers/msentraid/msentraid.go

```diff
@@ -39,6 +48,21 @@ func (p Provider) AuthOptions() []oauth2.AuthCodeOption {
 	return []oauth2.AuthCodeOption{}
 }
 
+func (p Provider) CheckTokenScopes(token *oauth2.Token) {
+	scopesStr, ok := token.Extra("scope").(string)
```

Comment:
I also don't suffix the type in general, but I do when I need to avoid a naming conflict. Here, we have the scopes first as a string and then as a slice. I prefer using the `Str` suffix in that case to make it clear that this is the only difference between the two. Using `tokenScopes` and `scopes` would imply some other difference IMO.

Reply:
I would really prefer we don’t use the pattern (we do it in some cases for the tests only which is really scoped to raw values we convert later on). If you count the whole Go stlib, most of them are in tests too and very few as variable names (mostly enum like). Could it be something like `rawScopes` (which sounds better than `scopesData`)?

Reply:
How is `rawScopes` better than `scopesStr`? I prefer descriptive names, and `Str` tells me more about the variable than `raw`.

Reply:
This is why I think `tokenScopes` was the most descriptive ones, the value that came directly from the token.

Reply:
IMO, variables should be named after their "purpose" rather than their type (and it's what we mainly see in other Go projects also). I'd go for `tokenScopes`.

Reply:
I found this in Google's Go style guide which supports including the underlying representation (`string` in this case) in the variable name:

https://google.github.io/styleguide/go/decisions.html#variable-name-vs-type

> If the value appears in multiple forms, this can be clarified either with an extra word like raw and parsed or with the underlying representation:
> 
> ```
> // Good:
> limitStr := r.FormValue("limit")
> limit, err := strconv.Atoi(limitStr)
> ```

It also mentions `raw` and `parsed`, which IMO make sense when the value is actually parsed or converted in some form, but here we just split a string, so I really prefer the `Str` suffix.